### PR TITLE
chore: populate lambda fn Code values without having to upload

### PR DESCRIPTION
**Pull request by efekarakus**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/cli/deploy/workload.go [hunk 1]
```diff
@@ -486,6 +491,26 @@ func (d *workloadDeployer) pushAddonsTemplateToS3Bucket() (string, error) {
...
+		return nil, err
+	}
+	// Assign defaults if not set.
+	if len(out.CustomResourcesURL) == 0 {
+		// Populate the custom resource paths that copilot should upload the resources to.
+		crs, err := d.customResources(d.templateFS)
+		if err != nil {
+			return nil, err
+		}
+		out.CustomResourcesURL = make(map[string]string)
```
Comment (iamhopaul123) [inline internal/pkg/cli/deploy/workload.go:505]: It feels odd to me to check if `len(out.CustomResourcesURL) == 0` and if so initialize `out.CustomResourcesURL` again. Isn't it empty already?

Comment (efekarakus) [reply internal/pkg/cli/deploy/workload.go:505]: ooh yea, the problem is that if `out.CustomResourcesURL` is nil then we can't write `out.CustomResourcesURL[cr.Name()] = ...`

Like I have to initialize the map if it's nil if that makes sense.

**Review by iamhopaul123 — APPROVED** [review]

File: internal/pkg/cli/deploy/workload.go [hunk 2]
```diff
@@ -486,6 +491,26 @@ func (d *workloadDeployer) pushAddonsTemplateToS3Bucket() (string, error) {
...
+		for _, cr := range crs {
+			out.CustomResourcesURL[cr.Name()] = s3.URL(d.resources.Region, d.resources.S3Bucket, cr.ArtifactPath())
```
Comment (Lou1415926) [inline internal/pkg/cli/deploy/workload.go:507]: What are your thoughts on handling all these in logic in `Template()` of `stack` package?

Comment (efekarakus) [reply internal/pkg/cli/deploy/workload.go:507]: That's a great idea, I like that!  Now I need to somehow find the time to do the refactor that moves the `customresources` from the `deploy` pkg to stack  :rip:

**Comment by efekarakus** [comment]
I'll close this PR in favor of Wanxian's recommendation which is a better approach: https://github.com/aws/copilot-cli/pull/4453/files#r1096275545

**Pull request closed without merging**